macroslib/src/lib.rs

```diff
@@ -377,6 +400,24 @@ impl Generator {
         }
     }
 
+    /// process string `src` and save result of macro expansion to `dst`
+    ///
+    /// # Panics
+    /// Panics on error
+    pub fn expand_from_str<D>(mut self, crate_name: &str, src: String, dst: D)
```

Comment:
Actually, it was added somewhat intenionally, for such case:
For the sake of cleaner project structure, we have more than one `rs.in` file. One of it depends on the other. The simplest way to support it was to concatenate them and pass to `rust_swig` as a string. It also enables custom preprocessing of a file, before it is passed into `rust_swig`.

That being said, it is completely different feature and was not supposed to be part of this PR. I 've removed the function. If you think it is useful, I can make separate PR later.

Reply:
> The simplest way to support it was to concatenate them and pass to rust_swig as a string. It also 
> enables custom preprocessing of a file, before it is passed into rust_swig.

It is funny, but removing `expand_from_str` was also step into this direction.
To report proper error related to multiply input source files rust_swig needs information about mapping file name = string with content, 
so `expand_str` now take SourceId instead of `&str` to support in future multiply calls.

With string concatenation it would be almost impossible to report user which line of source code
contains error.